lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/src/eos_helpers.cpp

```diff
@@ -190,223 +184,249 @@ namespace pew::eos
-
-        platform_options.EncryptionKey = eos_config.encryptionKey.length() > 0 ? eos_config.encryptionKey.c_str() : nullptr;
-        platform_options.OverrideCountryCode = eos_config.overrideCountryCode.length() > 0 ? eos_config.overrideCountryCode.c_str() : nullptr;
-        platform_options.OverrideLocaleCode = eos_config.overrideLocaleCode.length() > 0 ? eos_config.overrideLocaleCode.c_str() : nullptr;
-        platform_options.ProductId = eos_config.productID.c_str();
-        platform_options.SandboxId = eos_config.sandboxID.c_str();
-        platform_options.DeploymentId = eos_config.deploymentID.c_str();
-        platform_options.ClientCredentials.ClientId = eos_config.clientID.c_str();
-        platform_options.ClientCredentials.ClientSecret = eos_config.clientSecret.c_str();
-
-        platform_options.TickBudgetInMilliseconds = eos_config.tickBudgetInMilliseconds;
-
-        if (eos_config.taskNetworkTimeoutSeconds > 0)
+        // If the SteamAPI_Init function pointer is null, then it was not
+        // correctly retrieved from the steam library. Log an error and stop
+        if (SteamAPI_Init == nullptr)
         {
-            platform_options.TaskNetworkTimeoutSeconds = &eos_config.taskNetworkTimeoutSeconds;
+            logging::log_error(
+                "Could not load a pointer to the SteamAPI_Init "
+                "function within the loaded steam library handle.");
+            return;
         }
 
-        EOS_Platform_RTCOptions rtc_options = { 0 };
-
-        rtc_options.ApiVersion = EOS_PLATFORM_RTCOPTIONS_API_LATEST;
-#if PLATFORM_WINDOWS
-        logging::log_inform("setting up rtc");
-        std::filesystem::path xaudio2_dll_path = io_helpers::get_path_relative_to_current_module(XAUDIO2_DLL_NAME);
-        std::string xaudio2_dll_path_as_string = string_helpers::to_utf8_str(xaudio2_dll_path);
-        EOS_Windows_RTCOptions windows_rtc_options = { 0 };
-        windows_rtc_options.ApiVersion = EOS_WINDOWS_RTCOPTIONS_API_LATEST;
-        windows_rtc_options.XAudio29DllPath = xaudio2_dll_path_as_string.c_str();
-        logging::log_warn(xaudio2_dll_path_as_string.c_str());
-
-        if (!std::filesystem::exists(xaudio2_dll_path))
+        if (SteamAPI_Init())
         {
-            logging::log_warn("Missing XAudio dll!");
+            logging::log_inform("SteamAPI_Init returned true. "
+                "Steam has been initialized.");
         }
-        rtc_options.PlatformSpecificOptions = &windows_rtc_options;
-        platform_options.RTCOptions = &rtc_options;
-#endif
+        else
+        {
+            logging::log_error(
+                "SteamAPI_Init returned false. Steam was not able "
+                "to be initialized.");
+        }
+    }
 
-#if PLATFORM_WINDOWS
-        auto path_to_steam_config_json = config_legacy::get_path_for_eos_service_config(EOS_STEAM_CONFIG_FILENAME);
+    void apply_steam_settings(EOS_Platform_Options& platform_options) 
```

Comment:
Not for this review, but I have a style nitpick that I'd like to discuss with you about how to use references in the future.